service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ImmunizationObservationStatementMapper.java

```diff
@@ -177,12 +178,25 @@ private String buildExpirationDatePertinentInformation(Immunization immunization
     }
 
     private String buildSitePertinentInformation(Immunization immunization) {
-        Optional<String> site = Optional.empty();
-        if (immunization.hasSite()) {
-            CodeableConcept siteObject = immunization.getSite();
-            site = CodeableConceptMappingUtils.extractTextOrCoding(siteObject);
+        String siteUserSelected = immunization.getSite().getCoding().stream()
+            .filter(Coding::hasUserSelected)
+            .findFirst()
+            .map(Coding.class::cast)
+            .map(Coding::getDisplay)
+            .filter(StringUtils::isNotBlank)
+            .map(value -> SITE + value)
+            .orElse(StringUtils.EMPTY);
+
+        if (StringUtils.isBlank(siteUserSelected)) {
```

Comment:
This whole thing can be rewritten:
```
return immunization.getSite().getCoding().stream()
    .filter(Coding::hasUserSelected)
    .findFirst()
    .map(Coding.class::cast)
    .map(Coding::getDisplay)
    .filter(StringUtils::isNotBlank)
    .map(value -> SITE + value)
    .or(() -> Optional.of(immunization)
        .filter(Immunization::hasSite)
        .map(Immunization::getSite)
        .map(CodeableConceptMappingUtils::extractTextOrCoding)
        .map(value -> SITE + value))
    .orElse(StringUtils.EMPTY);
```

Reply:
Original code has been extracted into a new method in 'CodeableConceptMappingUtils'